Control plugin for Dahua PTZ cameras

jof opened the pull request.

knight-of-ni commented:
Thanks for including the line of sql that allows this to show up in the web ui. 

In various places we've got conflicting license statements in the code. See #817 for details.

Your two open PR's you have open have the same issue. It's certainly not your fault since you most likely used existing scripts as examples.  Since you identify yourself as the author of these control files, technically speaking only you can modify the license. Would you please modify your PR's to reflect only one license. Simplest thing would be to remove the Perl license statement at the bottom or change it to match the GPL license in the header.

connortechnology commented:
Please see https://github.com/ZoneMinder/zoneminder/blob/master/scripts/ZoneMinder/lib/ZoneMinder/Control/Trendnet.pm as a more up to date example.

jof commented:
Thanks for the reviews, folks!

I updated the license information and also added handling for http or https-style URIs, based on the comment that came up in #2191

connortechnology commented:
Looking good.  Nitpicking now: We don't need the license at top and bottom.  I recommend removing the license at the top.

I would like to see more consistent conformance to google code style but I won't be too hard on that.

jof commented:
No worries. Nitpicks are what keeps a codebase clean.

Top-license removed.

jof commented on scripts/ZoneMinder/lib/ZoneMinder/Control/Dahua.pm:
```diff
@@ -0,0 +1,366 @@
+    $self->{dahua_channel_number} = "0";
+
+    my ($protocol, $username, $password, $address)
+       = $self->{Monitor}->{ControlAddress} =~ /^(https?:\/\/)?([^:]+):([^\/@]+)@(.*)$/;
```
That's really good feedback. I guess there is no validation on the stored ControlAddress (since there can be so many applicable protocol schemes).

I used your scheme from Trendnet.pm in Dahua.pm. I'll also do the same change for #2191

Also, I removed some trailing whitespace from Trendnet.pm
